qt/scientific_interfaces/Indirect/test/ISISEnergyTransferModelTest.h

```diff
@@ -0,0 +1,605 @@
+    }
+  }
+
+  void testValidateRunDataAllValid() {
+    IETInputData inputData("iris26184_multi_graphite002_red");
+    IETConversionData conversionData(0.5, 1, 2);
+    IETGroupingData groupingData(IETGroupingType::DEFAULT);
+    IETBackgroundData backgroundData(false);
+    IETAnalysisData analysisData;
+    IETRebinData rebinData;
+    IETOutputData outputData;
+
+    IETRunData runData(inputData, conversionData, groupingData, backgroundData, analysisData, rebinData, outputData);
+
+    std::vector<std::string> errors = m_model->validateRunData(runData, 1, 10);
+    TS_ASSERT_EQUALS(errors.size(), 0);
+  }
+
+  void testValidatePlotDataInputInvalid() {
+    IETInputData inputData;
+    IETConversionData conversionData;
+    IETBackgroundData backgroundData;
+
+    IETPlotData plotData(inputData, conversionData, backgroundData);
+
+    std::vector<std::string> errors = m_model->validatePlotData(plotData);
+    TS_ASSERT_EQUALS(errors.size(), 1);
+    if (errors.size() == 1)
+      TS_ASSERT_EQUALS(errors[0], "You must select a run file.");
+  }
+
+  void testValidatePlotDataSpectraInvalid() {
+    IETInputData inputData("iris26184_multi_graphite002_red");
+    IETConversionData conversionData(0.5, 4, 2);
+    IETBackgroundData backgroundData(false);
+
+    IETPlotData plotData(inputData, conversionData, backgroundData);
+
+    std::vector<std::string> errors = m_model->validatePlotData(plotData);
+    TS_ASSERT_EQUALS(errors.size(), 1);
+    if (errors.size() == 1)
+      TS_ASSERT_EQUALS(errors[0], "Minimum spectra must be less than maximum spectra.");
+  }
+
+  void testValidatePlotDataBackgroundInvalid() {
+    IETInputData inputData("iris26184_multi_graphite002_red");
+    IETConversionData conversionData(0.5, 1, 2);
+    IETBackgroundData backgroundData(true, -1, 1);
+
+    IETPlotData plotData(inputData, conversionData, backgroundData);
+
+    std::vector<std::string> errors = m_model->validatePlotData(plotData);
+    TS_ASSERT_EQUALS(errors.size(), 2);
+    if (errors.size() == 2) {
+      TS_ASSERT_EQUALS(errors[0], "The Start of Background Removal is less than the minimum of the data range");
+      TS_ASSERT_EQUALS(errors[1], "The End of Background Removal is more than the maximum of the data range");
+    }
+  }
+
+  void testValidatePlotDataAllValid() {
+    IETInputData inputData("iris26184_multi_graphite002_red");
+    IETConversionData conversionData(0.5, 1, 2);
+    IETBackgroundData backgroundData(false);
+
+    IETPlotData plotData(inputData, conversionData, backgroundData);
+
+    std::vector<std::string> errors = m_model->validatePlotData(plotData);
+    TS_ASSERT_EQUALS(errors.size(), 0);
+  }
+
+private:
+  std::unique_ptr<IETModel> makeModel() { return std::make_unique<IETModel>(); }
+  IAlgorithm_sptr makeReductionAlgorithm() {
+    return AlgorithmManager::Instance().create("ISISIndirectEnergyTransferWrapper");
+  }
+
+  std::unique_ptr<IETModel> m_model;
+  IAlgorithm_sptr m_reductionAlg;
```

Comment:
I like these tests. Do we need tests for the other modes of rebin (e.g. log scale)?

Reply:
Which rebin mode? in the current UI it only supports "Single" and "Multiple"

Reply:
https://docs.mantidproject.org/v4.0.0/algorithms/Rebin-v1.html the rebin string determines the mode

Reply:
Ok, I will add more tests with different modes